catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java

```diff
@@ -1179,14 +1179,22 @@ protected UpdateResponse patchMetacards(List<MetacardChanges> metacardChanges)
       for (AttributeChange attributeChange : changeset.getAttributes()) {
         for (String id : changeset.getIds()) {
           List<String> values = attributeChange.getValues();
-          Metacard result = results.get(id).getMetacard();
+          Result result = results.get(id);
+          if (result == null) {
+            LOGGER.debug(
+                "Metacard {} either does not exist or user {} does not have permission to see it",
+                id,
+                subjectIdentifer);
+            throw new NotFoundException("Result was not found");
+          }
+          Metacard resultMetacard = result.getMetacard();
 
           Function<Serializable, Serializable> mapFunc = Function.identity();
-          if (isChangeTypeDate(attributeChange, result)) {
-            mapFunc = mapFunc.andThen(util::parseDate);
+          if (isChangeTypeDate(attributeChange, resultMetacard)) {
+            mapFunc = mapFunc.andThen(serializable -> Date.from(util.parseDate(serializable)));
```

Comment:
❓ Just want to confirm, @rzwiefel @andrewkfiedler or @djblue, temporal attributes should be placed on a metacard as `Date`, not `Instant` correct? It seems the signature of the util was changed but since both `Date` and `Instant` are `Serializable` the regression would have never been caught without tests.

Reply:
They should be `Date`s. https://github.com/codice/ddf/blob/master/catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/BasicTypes.java#L28